### ISSUE TITLE
File extension required in serverMiddleware

### ISSUE THREAD

**Issue by llienard** [nuxt/typescript#80]
**Describe the bug**
File extension .ts is required in nuxt.config.ts serverMiddleware

**To Reproduce**
1. Create a nuxt typerscript project
2. Create a file `server/middleware.ts`
3. Set it to nuxt.config.ts like 
```
{
serverMiddleware: [
    '~/server/middleware' // without extension
  ]
}
```
4. Build with `node ./node_modules/\@nuxt/typescript-runtime/bin/nuxt-ts.js build`
4. Start with `node ./node_modules/\@nuxt/typescript-runtime/bin/nuxt-ts.js start`
5. See error

Error: Cannot find module '~/server/middleware'

**Expected behavior**
This should work

**Comment by kevinmarrec** [comment]
@llienard I'm aware of this issue, thanks for creating it though !

The thing is that for now it's the `typescript-build` module that tells Nuxt looking for `.ts` extensions for serverMiddlewares. But as the module is not registered when using `nuxt-ts` start (cause it's a `buildModule`), it doesn't resolve `.ts` automatically.

Current workaround is moving `@nuxt/typescript-build` from `buildModules` to `modules` for now.
We're working on a feature around Nuxt core to be able to fix this issue.

**Comment by llienard** [comment]
Thanks for your reply @kevinmarrec 

I understand the issue but it would be nice to add a note to the documentation because the  `nuxt-ts dev` command works when server middlewares are set without extension but the `nuxt-ts start`  doesn't. It's very confusing :(

**Comment by kevinmarrec** [comment]
# Workarounds

**Workaround N°1** 
 Set `@nuxt/typescript-build` in `modules` instead of `buildModules`

**Workaround N°2 (*Recommended*)** 
 Force `ts` extension in `extensions` array of configuration :
```js
export default {
  extensions: ['ts']
}
```

**Workaround N°3** 
 Force `ts` extension in `extensions` array of configuration :
```js
export default {
  extensions: ['ts']
}

# Fix proposal

Use new CLI hooks (starting Nuxt 2.9.1), to make `nuxt-ts` ensure that `ts` extension is registered, even when using `nuxt-ts start`.

**Comment by llienard** [comment]
Hi, I have an other problem with alias path from typescript configuration.

When I import a file via an alias from a server middleware, I have an error on `nuxt-ts build`

```typescript
// server/middleware/request-logger.middleware.ts
import logger from '~/utils/ServerLogger'
// ...
```
Result :

![image](https://user-images.githubusercontent.com/49902526/64433109-208d3700-d0be-11e9-9f75-72f8e8e95098.png)

**Comment by kevinmarrec** [comment]
@llienard Should works if you use `.ts` extension everywhere.
Still it will be fixed with next Nuxt verison & https://github.com/nuxt/typescript/pull/97

**Comment by kevinmarrec** [comment]
Will be fixed with next runtime version and Nuxt 2.10 (there will be release note that force user to use Nuxt 2.10 to have the updated TS runtime working)

It will be fixed thanks to this refactor using Nuxt 2.10 hooks : https://github.com/nuxt/typescript/pull/97

Closing as the PR has been merged.

**Comment by andywampir** [comment]
> It seems new version doesn't works for me.
> I'm still getting error `Cannot find module: '~/server/router'`.
> 
> ### Versions
> `nuxt`: `2.10.2`
> `@nuxt/typescript-build`: `0.3.2`
> `@nuxt/typescript-runtime`: `0.2.2`
> `node`: `12.13.0`

I got help in Discord server.
Alias won't be recognized, cause it's runs outside webpack environment.
So in my case i need to use `import router from '../server/router'`.
